core/src/main/java/org/elasticsearch/index/mapper/GeoShapeFieldMapper.java

```diff
@@ -445,27 +447,41 @@ public Mapper parse(ParseContext context) throws IOException {
         try {
             Shape shape = context.parseExternalValue(Shape.class);
             if (shape == null) {
-                ShapeBuilder shapeBuilder = ShapeBuilder.parse(context.parser(), this);
+                ShapeBuilder shapeBuilder = ShapeParser.parse(context.parser(), this);
                 if (shapeBuilder == null) {
                     return null;
                 }
                 shape = shapeBuilder.build();
             }
-            if (fieldType().pointsOnly() && !(shape instanceof Point)) {
-                throw new MapperParsingException("[{" + fieldType().name() + "}] is configured for points only but a " +
+            if (fieldType().pointsOnly() == true) {
+                // index configured for pointsOnly
+                if (shape instanceof XShapeCollection && XShapeCollection.class.cast(shape).pointsOnly()) {
```

Comment:
Just out of curiosity, where does this piece of code come from? Is this merely a refactoring or new behaviour?

Reply:
Its technically a bug fix. Indexing would fail if `ShapeCollection` was a collection of `Point` types. That shouldn't happen on an index configured for `points_only : true`

Reply:
Since this already seems to be its own selfcontained commit, would if be difficult to pull it out into a separate PR then? Better to get its own comment etc. then.

Reply:
Yes, I like that idea. I'll open a separate PR and probably a separate issue for tracking purposes.